.github/workflows/base-test.yml

```diff
@@ -46,9 +51,19 @@ jobs:
           path: ${{ steps.go-cache-paths.outputs.go-mod }}
           key: ${{ runner.os }}-go-mod-base-test-${{ hashFiles('**/go.sum') }}
 
-      - name: Run Tests
-        run: go test ./...
+      - name: Run Tests (Non-Windows)
+        if: matrix.os != 'windows'
+        run: |
+          go test -v $(go list ./... | grep -v "/test/") ./... -timeout 45m
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run Windows Tests
+        if: matrix.os == 'windows'
+        run: |
+          # Run Windows-specific tests
+          go test -v -p 1 -tags windows -run ^TestWindows $(go list ./... | grep -v "/test/") -timeout 45m
```

Comment:
We should either remove windows testing from this workflow or fix the tests so that Windows tests can be run with `go test ./...`.